# Importing the Dataset

In [1]:
import pandas as pd
df = pd.read_csv("DBS_SingDollar.csv")

# Splitting the Variables into Dependent & Independent Variables

In [2]:
X = df.loc[:,["SGD"]]
Y = df.loc[:,["DBS"]]
print(X)

         SGD
0    1.38925
1    1.38707
2    1.38324
3    1.38378
4    1.37595
..       ...
117  1.44655
118  1.44814
119  1.45018
120  1.44954
121  1.44766

[122 rows x 1 columns]


# Creating the Linear Regression Model and Generating Predicted Results

In [3]:
from sklearn import linear_model
model = linear_model.LinearRegression()
model.fit(X,Y)
prediction = model.predict(X) #Stores the prediction into variable called prediction

# Checking the Model's Prediction Accuracy

In [4]:
from sklearn.metrics import mean_squared_error
rmse = mean_squared_error(Y,prediction)**0.5
print(rmse)
#print(rmse/Y.mean(axis = 0)*100)

0.6283698867042153


# Obtaining our model's cofficient and intercept

In [5]:
coefficient = model.coef_
intercept = model.intercept_
print(coefficient)
print(intercept)

[[-50.60094302]]
[90.22858515]


### Therefore, our model is DBS Share Price = -50.60094302SGD + 90.22858515